### ISSUE TITLE
Chemical pages

### ISSUE THREAD

**Issue by ValWood** [pombase/website#2200]
We would like chemical pages for reaction participants.

For example, 
<img width="1291" alt="Screenshot 2024-07-31 at 19 25 56" src="https://github.com/user-attachments/assets/7aaa54dc-e9d0-4e9d-93c4-3bcd868e94e8">

chemicals link to CHEBI. Instead we would like to link to a chemical page which list all of the  genes + reactions that the enzyme participates in (I haven't thought this through fully).

We also need to exclude what we refer to as "currency chemicals"  for example
phosphate,
diphosphate
H+
H2O
(where the pages would be ridiculously long).

The page would display something like this:

<img width="774" alt="Screenshot 2024-07-31 at 19 30 31" src="https://github.com/user-attachments/assets/c86157c8-8c8b-4d7b-a9ba-a258d158d01b">

but with a gene instead of a Rhea ID.

This would be incredibly useful  to:
- quickly identify the next steps in pathways
- more as we begin to chase up random chemicals that are not dealt with as parts of pathway curation (especially to identify specific detoxification candidates among the "unknown" proteins) .

I guess the first step  is to store the chemicals in Chado if we don't do that yet?

**Comment by ValWood** [comment]
This will also be very useful to identify when genes are branch points in pathways, we should be able to see quickly if a pathway is linear or at a branch point with multiple options. This will help us with decisions about 'starts' and 'ends' of pathways and  where to break up GO-CAM models.

**Comment by kimrutherford** [comment]
> I guess the first step is to store the chemicals in Chado if we don't do that yet?

We currently only store the chemicals from ChEBI that are in: `pombe-embl/mini-ontologies/chebi.obo`
The full ChEBI is 200,000 terms and we only use a handful.

I've had a first look at the Rhea downloads page.  I can't see a file the contains a simple mapping between the Rhea IDs and the ChEBI IDs.  There are RDF files but probably using the SPARQL endpoint would work: https://sparql.rhea-db.org/sparql/#

Something like this:
```sparql
PREFIX ch:<http://purl.obolibrary.org/obo/>

SELECT distinct ?chebi ?reaction
WHERE {
  ?reaction rdfs:subClassOf rh:Reaction .
  ?reaction rh:status rh:Approved .
  ?reaction rh:side ?reactionSide .
  ?reactionSide rh:contains ?participant .
  ?participant rh:compound ?compound .
  ?compound rh:chebi ?chebi .
  ?reaction rh:id "39051"^^xsd:long .

}
ORDER BY ?chebi
```

It will be slow to do many queries but it won't need updating every night.

**Comment by kimrutherford** [comment]
> chemicals link to CHEBI. Instead we would like to link to a chemical page

We'll need help from the widget developers for that because the widget code inserts the links to ChEBI automatically.

Would it be useful/necessary to search from chemicals with the quick search or in the query builder?

Currently the only reaction details we store are the Rhea IDs, which are dbxrefs for GO terms in Chado.  We can get more reaction details using SPARQL queries but I'm not sure of the best way to store the details in Chado (or if we should store the details in Chado at all).

Chemical pages could be a bit of work.  We won't be able to re-use much of the code from other pages because the table on the chemical pages with be unique to the new pages.

> but with a gene instead of a Rhea ID.

It could be multiple genes, I think, like in this case: https://www.pombase.org/term/GO:0004845

Would the Rhea ID and GO term ID(s) be a good idea in the table too?

Should we show the reaction diagrams on the chemical pages?

**Comment by ValWood** [comment]
Lets discuss on next group call and see if there are any ways to make this easier.
I'll think more about exactly what we need.

**Comment by kimrutherford** [comment]
We have (empty) term pages for chemicals already that we can build on:
https://www.pombase.org/term/CHEBI:142517

**Comment by PCarme** [comment]
A couple of ideas for what could go on these pages:
- Table with the GO MF terms in which the chemical is involved and all the genes annotated to these terms.
- For chemicals for which we have sensitivity phenotypes, lists of genotypes that are sensitive/resistant to these chemicals.
- For chemicals which affect the RNA/protein levels of some genes, the list of genes affected by the chemical.

**Comment by kimrutherford** [comment]
Thanks.

We should also be able to include the structure diagream from the ChEBI pages:

https://www.ebi.ac.uk/chebi/CHEBI:142517

<img width="300" height="300" alt="Image" src="https://github.com/user-attachments/assets/66ff3e73-4652-4774-8227-06ab3571799c" />

**Comment by ValWood** [comment]
sgd chemical page
https://www.yeastgenome.org/chemical/CHEBI:27732

**Comment by ValWood** [comment]
Might  want to consider  chemicals as conditions (although mot will be sensitivity phenotypes and so covered by the phenotypes)

**Comment by PCarme** [comment]
> Might want to consider chemicals as conditions (although mot will be sensitivity phenotypes and so covered by the phenotypes)

True, we may want to display all phenotypes that happen in presence of the chemical, not just sensitivity phenotypes

**Comment by kimrutherford** [comment]
> I've had a first look at the Rhea downloads page. I can't see a file the contains a simple mapping between the Rhea IDs and the ChEBI IDs.

I've just asked the Rhea helpdesk if there's a mapping file available.

**Comment by kimrutherford** [comment]
> I've just asked the Rhea helpdesk if there's a mapping file available.

Nicole Redaschi replied with a SPARQL query that can download a mapping file in CSV format.  I've put the results here:
`/var/pomcur/sources/rhea_chebi_id_mapping-2026-05-19.csv`

We'll need to update it occasionally.

The instructions from Rhea are on the wiki: https://github.com/pombase/pombase-chado/wiki/Updating-the-Rhea-to-CHEBI-mapping-file

**Comment by kimrutherford** [comment]
> Nicole Redaschi replied with a SPARQL query that can download a mapping file in CSV format. I've put the results here:
> /var/pomcur/sources/rhea_chebi_id_mapping-2026-05-19.csv

I've moved it to SVN:
`pombe-embl/external_data/rhea_chebi_id_mapping.csv`
and added the loading step to read the mapping into Chado.

**Comment by kimrutherford** [comment]
> We currently only store the chemicals from ChEBI that are in: pombe-embl/mini-ontologies/chebi.obo
> The full ChEBI is 200,000 terms and we only use a handful.

We might need to rethink this.  There are are only 165 chemicals in our mini ChEBI file but there are thousands of chemicals in the Rhea reactions that we have on the term pages.  I thought the easy plan was to make a chemical page for each ChEBI term in Chado, but there are only the 165 from the mini-OBO file.

We'll need to do something cleverer like reading the full ChEBI OBO file, but only loading a filtered set of terms (those in Rhea reactions).

**Comment by ValWood** [comment]
I also have a lot of chemicals to add to the imports so I can create logical definitions. I will spend a day on that soon.

**Comment by kimrutherford** [comment]
Something else to discuss:
Some chemicals appear in a lot of Rhea reactions.  For example pyruvate appears in over 200 so we might not want to show the reactions on the pyruvate page.
